crypto/crypto/src/hash.rs

```diff
@@ -605,7 +612,104 @@ impl<T: ser::Serialize + ?Sized> TestOnlyHash for T {
     fn test_only_hash(&self) -> HashValue {
         let bytes = lcs::to_bytes(self).expect("serialize failed during hash.");
         let mut hasher = TestOnlyHasher::default();
-        hasher.write(&bytes);
+        hasher.update(&bytes);
+        hasher.finish()
+    }
+}
+
+/// Alternative to CryptoHash
+pub trait SimpleHash {
+    /// The hash method
+    fn hash(&self) -> HashValue;
```

Comment:
Whether to consider using an alternative name to avoid conflict with the std Hash trait?

Reply:
I remember a long discussion where I was supporting `digest` for this reason, but I'm not sure I want to re-open the question :)